src/instruments/src/BAT/index.tsx

```diff
@@ -1,14 +1,76 @@
 import './style.scss';
-import React from 'react';
+import React, { useCallback, useRef, useState } from 'react';
 import { render } from '@instruments/common/index';
 import { useSimVar } from '@instruments/common/simVars';
+import { useUpdate } from '@instruments/common/hooks';
 
 const BatDisplay = ({ number, x, y }) => {
-    const [ltsTest] = useSimVar('L:A32NX_OVHD_INTLT_ANN', 'Bool', 200);
-    const [voltage] = useSimVar(`L:A32NX_ELEC_BAT_${number}_POTENTIAL`, 'Volts', 200);
+    const [ltsTest] = useSimVar('L:A32NX_OVHD_INTLT_ANN', 'Bool', 100);
+    const [voltage] = useSimVar(`L:A32NX_ELEC_BAT_${number}_POTENTIAL`, 'Volts', 100);
+    const [digits, setDigits] = useState<string>('   ');
+    const updated = useRef(0); // how many of the digits have already been updated
+    const timer = useRef(0); // how much time has passed since we started over last
+
+    /**
+     * Updates a single digit. No dependencies are necessary as we use a setter callback
+     */
+    const setDigit = useCallback((index: number, digit: string) => {
+        setDigits((oldDigits) => `${oldDigits.slice(0, index)}${digit}${oldDigits.slice(index + 1)}`);
+    }, []);
+
+    const voltageValue = voltage.toFixed(1); // prevent unnecessary rebuilds of getDisplayValue
+    const getDisplayValue: () => string = useCallback(() => {
+        if (ltsTest === 0) {
+            return '888';
+        }
+
+        const int = Math.floor(voltageValue);
+        const decimal = (voltageValue - int).toFixed(1).substring(2);
+
+        let intDigits = int.toString();
+        while (intDigits.length < 2) {
+            intDigits = `0${intDigits}`;
```

Comment:
Just a reflection, may be more useful to treat digits as a char array rather than as a string, as when it is accessed for rendering index value is often referenced

Reply:
Although this already flows quite nicely so maybe only refactor later on if it proves to be an issue.